Add midjourney extension

CarterMcAlister opened the pull request:
## Description

This extension allows users to interact with Midjourney directly from Raycast via making calls to the Discord api.

### Features
- Generate new images with Midjourney from a prompt
- View images you've created with Midjourney
- Remix and upscale generated images with Midjourney
- Export the images you create to clipboard or your filesystem

## Screencast
![midjourney-1](https://github.com/raycast/extensions/assets/5679378/81327338-08c6-42eb-aedb-8f9943385955)
![midjourney-2](https://github.com/raycast/extensions/assets/5679378/d8608824-248c-4771-8406-19446a05c31a)
![midjourney-3](https://github.com/raycast/extensions/assets/5679378/0a0b4ecd-f132-4e96-8007-34d63c642c0c)
![midjourney-4](https://github.com/raycast/extensions/assets/5679378/33f15a74-c3c2-4c08-a6a2-d0992c8c486d)
![midjourney-5](https://github.com/raycast/extensions/assets/5679378/d2cccce6-a1d2-4271-bdb4-ec72f7e8f2c7)
![midjourney-6](https://github.com/raycast/extensions/assets/5679378/1afe43f5-4ae3-4217-b527-45df499ca231)
![midjourney-7](https://github.com/raycast/extensions/assets/5679378/4e0d5ac9-d868-47a6-9c9f-553709b5ba61)

## Checklist
- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- The extension crashes if you have the wrong credentials, could you handle it gracefully?

![midjourney 2023-10-12 at 11 23 39](https://github.com/raycast/extensions/assets/38033107/f925f7cb-611e-4e7d-b968-491f85627f51)

- You don’t need to have the same issues in a folder called media if they are 100% the same, you can reference metadata 

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

CarterMcAlister commented:
@pernielsentikaer thanks for the feedback!
I have added error handling for invalid credentials, and removed the media folder.
Ready for a new review at your convenience
